src/_pytest/config/__init__.py

```diff
@@ -1305,6 +1316,69 @@ def _getconftest_pathlist(
             values.append(relroot)
         return values
 
+    def _get_append_ini_values(self, name: str, ini_type) -> List[str]:
+        value = []
+        # append_ini is a list of "ini=value" options
+        # append all values if multiple values are set for same ini-name,
+        # e.g. -a foo=bar1 -a foo=bar2 will append 'bar1 bar2' to foo
+        for ini_config in self._append_ini:
+            try:
+                key, append_ini_value = ini_config.split("=", 1)
+            except ValueError:
+                raise UsageError(
+                    "-a/--append-ini expects option=value style (got: {!r}).".format(
+                        ini_config
+                    )
+                )
+            else:
+                if key == name:
+                    if ini_type in ["pathlist", "args"]:
+                        value.append(append_ini_value)
+                    elif ini_type == "linelist":
+                        value.append("\n{}".format(append_ini_value))
+                    else:
+                        self._warn_or_fail_if_strict(
+                            "append_ini option invalid for argument '{}' with type '{}'".format(
+                                name, ini_type
+                            )
+                        )
+        return value
+
+    def _resolve_ini_option(
+        self,
+        name: str,
+        default_value: str,
+        override_value: Optional[str],
+        append_values: List,
+        ini_type: Optional[str],
+    ) -> Union[List, str, None]:
+        value = self.inicfg.get(name, None)
+        if override_value is None and not append_values:
+            if value is None:
+                if default_value is not None:
+                    return default_value
+                if ini_type is None:
+                    return ""
+                return []
+        else:
+            if override_value:
+                value = override_value
+            if append_values:
+                if not value:
+                    fail(
+                        "append_ini option invalid for argument '{}' since it has no value".format(
+                            name
+                        ),
+                        pytrace=False,
+                    )
+                else:
+                    value = (
+                        value + append_values
+                        if isinstance(value, list)
+                        else "{} {}".format(value, " ".join(append_values))
```

Comment:
I think maybe it will be better to write the code such that the appended values are only added the base value is coerced. Then this joining won't be necessary. I feel this is a bit prone to error.

Reply:
I'm -1 on that only because we'd have to perform the coercion logic twice then. Once for the base values and once for the appended values. Also, this way we make it so that we treat the appended values the same way as the base values. If we don't we risk opening up room for bugs since they're regarded as two seperate sets of entities when really they're on large set ( values = values + appended )

Reply:
OK :+1: